### PR TITLE
feat: Allow manually set clustering_columns in dataframe.to_gbq

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: tests/system/small/test_dataframe_io.py [hunk 1]
```diff
@@ -317,6 +317,51 @@ def test_to_gbq_w_None_column_names(
     )
 
 
+@pytest.mark.parametrize(
+    "clustering_columns",
+    [
+        pytest.param(["int64_col", "geography_col"]),
+        pytest.param(
+            ["float64_col"],
+            marks=pytest.mark.xfail(raises=google.api_core.exceptions.BadRequest),
```
Comment (shobsi) [inline tests/system/small/test_dataframe_io.py:326]: Worth adding one more testcase with duplicate clustering column names throwing the exception

File: bigframes/dataframe.py [hunk 2]
```diff
@@ -2756,6 +2767,52 @@ def _apply_unary_op(self, operation: ops.UnaryOp) -> DataFrame:
...
+            return mapped_columns
+
+        if not clustering_columns:
+            return []
+
+        if len(list(clustering_columns)) != len(set(clustering_columns)):
+            raise ValueError("Duplicates are not supported in clustering_columns")
+
+        all_possible_columns = (
+            set(self.columns) | set(self.index.names) if index else set(self.columns)
```
Comment (shobsi) [inline bigframes/dataframe.py:2792]: [nit] not clear if `|` appplies first or the `if-else`, maybe an explicit parentheses would be nice

File: tests/system/small/test_dataframe_io.py (hunk 1 above)
Comment (Genesis929) [reply tests/system/small/test_dataframe_io.py:326]: Done.

File: bigframes/dataframe.py (hunk 2 above)
Comment (Genesis929) [reply bigframes/dataframe.py]: Done.

**Review by shobsi — APPROVED** [review]

**Pull request merged**